apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/main/java/gov/cms/bfd/pipeline/ccw/rif/extract/s3/DataSetManifestFactory.java

```diff
@@ -0,0 +1,106 @@
+
+/**
+ * This singleton class defines the XML parsing functionality specific to a BFD file manifest; it is
+ * used to assert XML <code>manifest.xml</code> files are both well-formed and have valid content.
+ * The embedded XML Schema Definition (XSD) will be specified per JAXB schema validation to ensure
+ * data is valid.
+ */
+public class DataSetManifestFactory {
+  /** The {@link String} pipeline manifest XML Schema Definition (XSD). */
+  public static final String MANIFEST_XSD =
+      "<?xml version=\"1.0\" encoding=\"utf-8\"?>"
+          + "<xs:schema attributeFormDefault=\"unqualified\" elementFormDefault=\"qualified\""
+          + "	targetNamespace=\"http://cms.hhs.gov/bluebutton/api/schema/ccw-rif/v9\""
+          + "	xmlns:xs=\"http://www.w3.org/2001/XMLSchema\">"
+          + "  <xs:element name=\"dataSetManifest\">"
+          + "    <xs:complexType>"
+          + "      <xs:sequence>"
+          + "        <xs:element maxOccurs=\"unbounded\" name=\"entry\">"
+          + "          <xs:complexType>"
+          + "            <xs:attribute name=\"name\" type=\"xs:string\" use=\"required\" />"
+          + "            <xs:attribute name=\"type\" type=\"xs:string\" use=\"required\" />"
+          + "          </xs:complexType>"
+          + "        </xs:element>"
+          + "      </xs:sequence>"
+          + "      <xs:attribute name=\"timestamp\" type=\"xs:dateTime\" use=\"required\" />"
+          + "      <xs:attribute name=\"sequenceId\" type=\"xs:unsignedByte\" use=\"required\" />"
+          + "      <xs:attribute name=\"syntheticData\" type=\"xs:boolean\" use=\"optional\" />"
+          + "    </xs:complexType>"
+          + "  </xs:element>"
+          + "</xs:schema>";
+
+  /** The {@link DataSetManifestFactory} our factory instance (singleton). */
+  private static DataSetManifestFactory thisInstance;
+
+  /** private constructor...only we get to use it {@link DataSetManifestFactory}. */
+  private DataSetManifestFactory() {}
+
+  /**
+   * Constructs a new {@link DataSetManifestFactory} instance.
+   *
+   * @return the {@link DataSetManifestFactory}
+   */
+  public static DataSetManifestFactory newInstance() {
+    if (thisInstance == null) {
+      thisInstance = new DataSetManifestFactory();
+    }
+    return thisInstance;
+  }
+
+  /**
+   * Parses the input filename {@link String} and returns a new {@link DataSetManifest} instance.
+   *
+   * @param manifestFilename the {@link String} XML filename to read/parse
+   * @return the {@link DataSetManifest}
+   * @throws JAXBException (any errors encountered will be bubbled up).
+   */
+  public DataSetManifest parseManifest(String manifestFilename) throws JAXBException {
+    try {
+      InputStream manifestStream =
+          Thread.currentThread().getContextClassLoader().getResourceAsStream(manifestFilename);
+      return parseManifest(manifestStream);
+    } catch (Exception e) {
+      throw new JAXBException(e.getMessage(), e);
```

Comment:
I think we should avoid constructing JAXBExceptions ourselves -- it could be confusing for debugging purposes since most will assume that JAXBExceptions originate from JAXB.

Reply:
The pipeline is built to 'exit gracefully' (?oxymoron) on a JAXBException.....the AC was written to preserve that concept....

During unmarshalling, at least 3 diff exceptions are possible:

1. SAXException occurs when the syntheticData attribute value is validated vs. the XSD.
2. UnmarshallException when the unmarshall op fails
3. JAXBException - malformed XML, etc.

Since the first two are not currently handled by the pipeline AND the pipeline exception handling / shutdown is tied to look for a JAXBException, I left things as they were in the pipeline per the AC.

Reply:
changed RifLoader to handle both JAXBException and SAXException.